### ISSUE TITLE
Crash on iOS when generating addresses

### ISSUE THREAD

**Issue by rajivshah3** [iotaledger/trinity-wallet#1339]
### Bug description
We're receiving reports of crashes occurring on iOS when generating addresses or signatures

### Environment Specs
What device are you running Trinity on?

- Manufacturer: Apple
- Model: iPhone 7, iPhone X
- OS version: iOS 12.2 (16E227)

What build of Trinity are you running? Please provide the full version (you can find it in Settings > About Trinity).

- Trinity version: v0.6.2 (54)

### Steps To Reproduce
Run snapshot transition, manual sync, or generate an address on the Receive tab.

### Expected behaviour
Features should run as expected and not crash

### Actual behaviour
Crash occurred

### Errors
<details>
<summary>Crash report (iPhone X)</summary>

```

Thread 4 Crashed:
0   libsystem_kernel.dylib        	0x00000001d8e120dc __pthread_kill + 8
1   libsystem_pthread.dylib       	0x00000001d8e8f9b0 pthread_kill$VARIANT$armv81 + 296
2   libsystem_c.dylib             	0x00000001d8d6bf4c __abort + 152
3   libsystem_c.dylib             	0x00000001d8d6c770 a64l + 0
4   iotaWallet                    	0x0000000102cea170 kerl_absorb + 188
5   iotaWallet                    	0x0000000102ce97b0 iss_kerl_subseed + 76
6   iotaWallet                    	0x0000000102ce8b18 iota_sign_address_gen_trits + 172
7   iotaWallet                    	0x0000000102ce653c +[EntangledIOSBindings iota_ios_sign_address_gen_trits:index:security:] + 56
8   iotaWallet                    	0x00000001027b2048 -[EntangledIOS generateAddress:index:security:resolver:rejecter:] + 196 (EntangledIOS.m:50)
9   CoreFoundation                	0x00000001d927bba0 __invoking___ + 144
10  CoreFoundation                	0x00000001d915dc90 -[NSInvocation invoke] + 292
11  CoreFoundation                	0x00000001d915e8c4 -[NSInvocation invokeWithTarget:] + 60
12  iotaWallet                    	0x0000000102b6748c -[RCTModuleMethod invokeWithBridge:module:arguments:] + 472
13  iotaWallet                    	0x0000000102bacc84 facebook::react::invokeInner(RCTBridge*, RCTModuleData*, unsigned int, folly::dynamic const&) + 248
14  iotaWallet                    	0x0000000102bac9e0 invocation function for block in facebook::react::RCTNativeModule::invoke(unsigned int, folly::dy... + 88
15  libdispatch.dylib             	0x00000001d8cb4a38 _dispatch_call_block_and_release + 24
16  libdispatch.dylib             	0x00000001d8cb57d4 _dispatch_client_callout + 16
17  libdispatch.dylib             	0x00000001d8c90dec _dispatch_lane_serial_drain$VARIANT$armv81 + 548
18  libdispatch.dylib             	0x00000001d8c9192c _dispatch_lane_invoke$VARIANT$armv81 + 408
19  libdispatch.dylib             	0x00000001d8c99e08 _dispatch_workloop_worker_thread + 584
20  libsystem_pthread.dylib       	0x00000001d8e95114 _pthread_wqthread + 304
21  libsystem_pthread.dylib       	0x00000001d8e97cd4 start_wqthread + 4

```

</details>

**Comment by rajivshah3** [comment]
There was also a similar crash on an iPhone XS (iOS 12.1.4) in build 52

<details>
<summary>Crash report</summary>

```

Thread 3 Crashed:
0   libsystem_kernel.dylib        	0x00000001e62f50e4 __pthread_kill + 8
1   libsystem_c.dylib             	0x00000001e624c058 __abort + 156 (abort.c:131)
2   libsystem_c.dylib             	0x00000001e624c8b4 __stack_chk_fail + 212 (stack_protector.c:140)
3   iotaWallet                    	0x0000000105157d48 kerl_absorb + 188
4   iotaWallet                    	0x0000000105157518 iss_kerl_subseed + 80
5   iotaWallet                    	0x0000000105156b8c iota_sign_address_gen_trits + 180
6   iotaWallet                    	0x0000000105154700 +[EntangledIOSBindings iota_ios_sign_address_gen_trits:index:security:] + 56
7   iotaWallet                    	0x0000000104c0c76c __73-[EntangledIOS generateAddresses:index:security:total:resolver:rejecter:]_block_invoke + 216 (EntangledIOS.m:76)
8   libdispatch.dylib             	0x00000001e616bb9c _dispatch_call_block_and_release + 32 (init.c:1372)
9   libdispatch.dylib             	0x00000001e616d134 _dispatch_client_callout + 20 (object.m:511)
10  libdispatch.dylib             	0x00000001e616fd88 _dispatch_queue_override_invoke + 668 (inline_internal.h:2441)
11  libdispatch.dylib             	0x00000001e617c04c _dispatch_root_queue_drain + 348 (inline_internal.h:2482)
12  libdispatch.dylib             	0x00000001e617c8b8 _dispatch_worker_thread2 + 120 (queue.c:6072)
13  libsystem_pthread.dylib       	0x00000001e636ebc0 _pthread_wqthread + 476 (pthread.c:2361)
14  libsystem_pthread.dylib       	0x00000001e6374dd4 start_wqthread + 4

```

</details>

**Comment by rajivshah3** [comment]
Fixed in #1674 and #1741
